erpnext/controllers/buying_controller.py

```diff
@@ -437,17 +437,20 @@ def validate_purchase_return(self):
 			# validate rate with ref PR
 
 	def validate_rejected_warehouse(self):
-		for d in self.get("items"):
-			if flt(d.rejected_qty) and not d.rejected_warehouse:
+		for item in self.get("items"):
+			if flt(item.rejected_qty) and not item.rejected_warehouse:
 				if self.rejected_warehouse:
-					d.rejected_warehouse = self.rejected_warehouse
+					item.rejected_warehouse = self.rejected_warehouse
 
-				if not d.rejected_warehouse:
-					frappe.throw(
-						_("Row #{0}: Rejected Warehouse is mandatory against rejected Item {1}").format(
-							d.idx, d.item_code
-						)
+				if not item.rejected_warehouse:
+					msg = (
+						f"Row #{item.idx}: Rejected Warehouse is mandatory against rejected Item {item.item_code}"
 					)
+					frappe.throw(_(msg), title=_("Mandatory Rejected Warehouse"))
+
+			if item.get("rejected_warehouse") and (item.get("rejected_warehouse") == item.get("warehouse")):
+				msg = f"Row #{item.idx}: Accepted Warehouse and Rejected Warehouse cannot be same"
+				frappe.throw(_(msg), title=_("Invalid Warehouse"))
```

Comment:
These error messages will be untranslatable. `_("{0} {1}").format(p1, p2)` was the correct syntax.

Reply:
Fixed

Reply:
I don't see this change. Basically, we cannot use f-strings for translatable messages. We need to first call the translation function, then call `.format()` on the result

Reply:
Fixed